superset/assets/src/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -95,18 +95,9 @@ class SqlEditorLeftBar extends React.PureComponent {
       this.setState({ tableName: '' });
       return;
     }
-    const namePieces = tableOpt.value.split('.');
-    let tableName = namePieces[0];
-    let schemaName = this.props.queryEditor.schema;
-    if (namePieces.length === 1) {
-      this.setState({ tableName });
-    } else {
-      schemaName = namePieces[0];
-      tableName = namePieces[1];
-      this.setState({ tableName });
-      this.props.actions.queryEditorSetSchema(this.props.queryEditor, schemaName);
-      this.fetchTables(this.props.queryEditor.dbId, schemaName);
-    }
```

Comment:
The current code allows switching schema by typing `new_schema.some_table` in the table selector. The code breaks when the table name has a period in it, since it does naive splitting. I opted for removing it here, since it seems a bit unintuitive as well.